Statusseite und Config-Mode: CSS und JavaScript mit Versionsanhang laden.

Die Dateien unter /static/ werden ohne Cache-Control ausgeliefert, und ihr
Last-Modified ist bei jedem Build gleich (OpenWrt schreibt die Zeitstempel
fest, hier 29.11.2025). Browser schaetzen daraus eine Haltbarkeit von
grob 10 % des Dateialters - Wochen - und fragen in der Zeit nicht nach. Nach
einem Firmware-Update liefen so altes JavaScript und altes CSS weiter:

  * Statusseite: das neue Format nfLink (statuspage-respondd) fehlte im
    alten status-page.js, die Ethernet-Zellen blieben stehen, bis Strg+F5.
  * Config-Mode (wieder 192.168.1.1): neues gluon.css, alter Cache. Die
    Ansage an Aufsteller war deshalb bisher "Router-Config im neuen privaten
    Browserfenster".

Jetzt haengt an jeder Einbindung ?v=<Firmware-Release> aus
/lib/gluon/release. Jede Firmware hat damit eigene Adressen, der Browser
laedt nach einem Update frisch. uhttpd liefert die Datei unabhaengig vom
Query-String aus. Betroffen:

  status-page.html  status-page.css, status-page.js
  layout.html       status-page.css (Fehlerseiten der Statusseite)
  theme/layout.html gluon.css (Config-Mode)
  model/wrapper.html gluon-web-model.js (Formulare im Config-Mode)
  model/osm/map.html gluon-web-osm.js (Karte im Config-Mode)

Setzt auf statuspage-respondd auf (Kontext in status-page.html).

diff --git a/package/gluon-config-mode-theme/files/lib/gluon/config-mode/view/theme/layout.html b/package/gluon-config-mode-theme/files/lib/gluon/config-mode/view/theme/layout.html
index 9633531..b1ef0b9 100644
--- a/package/gluon-config-mode-theme/files/lib/gluon/config-mode/view/theme/layout.html
+++ b/package/gluon-config-mode-theme/files/lib/gluon/config-mode/view/theme/layout.html
@@ -95,7 +95,7 @@ SPDX-FileCopyrightText: 2008-2010 Jo-Philipp Wich <xm@subsignal.org>
 <html lang="">
 	<head>
 		<meta charset="UTF-8">
-		<link rel="stylesheet" type="text/css" media="screen" href="/static/gluon.css">
+		<link rel="stylesheet" type="text/css" media="screen" href="/static/gluon.css?v=<%| ((require('gluon.util').readfile('/lib/gluon/release') or ''):gsub('%s+', '')) %>">
 		<title><%| hostname .. ((rnode and rnode.title) and ' - ' .. title(rnode) or '') %></title>
 	</head>
 	<body>
diff --git a/package/gluon-status-page/files/lib/gluon/status-page/view/layout.html b/package/gluon-status-page/files/lib/gluon/status-page/view/layout.html
index d3201ef..bcc330b 100644
--- a/package/gluon-status-page/files/lib/gluon/status-page/view/layout.html
+++ b/package/gluon-status-page/files/lib/gluon/status-page/view/layout.html
@@ -10,7 +10,7 @@
 
 		<title><%:Error%></title>
 
-		<link rel="stylesheet" href="/static/status-page.css" type="text/css">
+		<link rel="stylesheet" href="/static/status-page.css?v=<%| ((require('gluon.util').readfile('/lib/gluon/release') or ''):gsub('%s+', '')) %>" type="text/css">
 	</head>
 	<body>
 		<header>
diff --git a/package/gluon-status-page/files/lib/gluon/status-page/view/status-page.html b/package/gluon-status-page/files/lib/gluon/status-page/view/status-page.html
index 725ca96..bc0c45f 100644
--- a/package/gluon-status-page/files/lib/gluon/status-page/view/status-page.html
+++ b/package/gluon-status-page/files/lib/gluon/status-page/view/status-page.html
@@ -297,7 +297,7 @@
 
 		<title><%| nodeinfo.hostname %> - <%:Status%></title>
 
-		<link rel="stylesheet" href="/static/status-page.css" type="text/css">
+		<link rel="stylesheet" href="/static/status-page.css?v=<%| ((require('gluon.util').readfile('/lib/gluon/release') or ''):gsub('%s+', '')) %>" type="text/css">
 	</head>
 	<body data-node-address="<%| http:getenv('SERVER_ADDR') %>"<%=
 		attr('data-translations', translations) ..
@@ -530,6 +530,6 @@
 				%>
 			</div>
 		</div>
-		<script src="/static/status-page.js"></script>
+		<script src="/static/status-page.js?v=<%| ((require('gluon.util').readfile('/lib/gluon/release') or ''):gsub('%s+', '')) %>"></script>
 	</body>
 </html>
diff --git a/package/gluon-web-model/files/lib/gluon/web/view/model/wrapper.html b/package/gluon-web-model/files/lib/gluon/web/view/model/wrapper.html
index a49fe93..96600a9 100644
--- a/package/gluon-web-model/files/lib/gluon/web/view/model/wrapper.html
+++ b/package/gluon-web-model/files/lib/gluon/web/view/model/wrapper.html
@@ -3,4 +3,4 @@
 		map:render(renderer)
 	end
 %>
-<script type="text/javascript" src="/static/gluon-web-model.js"></script>
+<script type="text/javascript" src="/static/gluon-web-model.js?v=<%| ((require('gluon.util').readfile('/lib/gluon/release') or ''):gsub('%s+', '')) %>"></script>
diff --git a/package/gluon-web-osm/files/lib/gluon/web/view/model/osm/map.html b/package/gluon-web-osm/files/lib/gluon/web/view/model/osm/map.html
index 2caa3f3..67c119c 100644
--- a/package/gluon-web-osm/files/lib/gluon/web/view/model/osm/map.html
+++ b/package/gluon-web-osm/files/lib/gluon/web/view/model/osm/map.html
@@ -1,5 +1,5 @@
 <div id="<%=id%>" class="gluon-osm-map" style="display: none"></div>
-<script type="text/javascript" src="/static/gluon-web-osm.js"></script>
+<script type="text/javascript" src="/static/gluon-web-osm.js?v=<%| ((require('gluon.util').readfile('/lib/gluon/release') or ''):gsub('%s+', '')) %>"></script>
 <script type="text/javascript">
 	(function() {
 		var elMap = document.getElementById(<%=json(id)%>);
